packages/fetchai/skills/ml_train/tasks.py

```diff
@@ -35,22 +36,38 @@ def __init__(
         self,
         skill_context: SkillContext,
         train_data: Tuple[np.ndarray, np.ndarray],
-        model: keras.Model,
         epochs_per_batch: int = 10,
         batch_size: int = 32,
     ):
         """Initialize the task."""
         super().__init__(logger=skill_context.logger)
         self.train_x, self.train_y = train_data
 
-        self.model = model
+        self.model = self._make_model()
         self.epochs_per_batch = epochs_per_batch
         self.batch_size = batch_size
 
     def setup(self) -> None:
         """Set up the task."""
         self.logger.info("ML Train task: setup method called.")
 
+    @staticmethod
+    def _make_model() -> Any:
+        """Make the model."""
+        model = tf.keras.Sequential(
+            [
+                tf.keras.layers.Flatten(input_shape=(28, 28)),
+                tf.keras.layers.Dense(128, activation="relu"),
+                tf.keras.layers.Dense(10, activation="softmax"),
+            ]
+        )
+        model.compile(
+            optimizer="adam",
+            loss="sparse_categorical_crossentropy",
+            metrics=["accuracy"],
+        )
+        return model
```

Comment:
Where was this done previously?

Reply:
in the model, so model spawned a thread and do the calculations on update method.